Products/ZenModel/UserSettings.py

```diff
@@ -1151,6 +1151,29 @@ def manage_deleteAdministrativeRole(self, delids=(), REQUEST=None):
                 )
             return self.callZenScreen(REQUEST)
 
+    security.declareProtected(ZEN_CHANGE_SETTINGS, 'getAllAdminGuids')
+    def getAllAdminGuids(self, returnChildrenForRootObj=False):
+        """
+        Return all guids user has permissions to.
+        """
+        guids = []
+        ars = self.getAllAdminRoles()
+        for ar in ars:
+            if not returnChildrenForRootObj:
+                guids.append(IGlobalIdentifier(ar.managedObject()).getGUID())
+            else:
```

Comment:
The `for` loop and this `if` statement regarding the `returnChildrenForRootObj` should be inverted.  The result of the `if` is the same regardless of the value of `ar`.  Granted, there will be two loops, but they will be less complicated.  E.g.
```
if not returnChildrenForRootObj:
    guids.extend(
        IGlobalIdentifier(ar.managedObject()).getGUID()
        for ar in ars
    )
else:
    for ar in ars:
        if ar.managedObject().getPrimaryId() in ....
```